### PR TITLE
Features/complete graph spec 178435322

### LINKED ISSUE #86: Complete Graph Spec Section
The Graph section of the spec needs to be written.

We have the actual graph data in the batch messages, but nothing about how to read, compile, or use the graph. This should go in the graph section of the spec https://spec.projectliberty.io/Graph/Overview

### PR DESCRIPTION
**Pull request by aneyzberg**
Problem
=======
Add graph Spec
[link to Pivotal Tracker #178435322](https://www.pivotaltracker.com/story/show/178435322)

Solution
========
Add Graph section to the spec. Explain how to retrieve compile and read the graph.

Closes #86

### CODE REVIEW

**Comment by aneyzberg** [comment]
How are we referring to DSNP identities for short? Are they DSNP Ids?

File: pages/Graph/Overview.md [hunk 1]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
 | 0.1     | Draft |
 
 ## Purpose
-
-TBD
+1. Describe the form of Graph Message posted to the blockchain
+1. Explain how to retrieve and resolve the graph
```
Comment (amparks100) [inline pages/Graph/Overview.md:24]: Should these be different numbers?

File: pages/Graph/Overview.md [hunk 2]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+An "unfollow" is the act of publicly unfollowing a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in the removal of this DSNP Id from a user's social graph
+
+### Graph
 
-## Details
+The collection of DSNP identities that a given user is following via the DSNP protocol. 
 
+#### Public
+Publicly following or unfollowing a DSNP Id implies that anyone looking for these follow or unfollow events on the blockchain will
+be able to determine who the follower(DSNP id doing the following) and followee(DSNP Id being followed) are for a given event.
```
Comment (amparks100) [inline pages/Graph/Overview.md:52]: Follower Id should be capitalized like followee Id

File: pages/Graph/Overview.md [hunk 3]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+
+To retrieve the graph - do the following:
+1. Retrieve all the log events[dsnpType](/Messages/Types)`GraphChange` from the chain
+1. Retrieve the batch file from each log event. Each log event of type [GraphChange](/Messages/Announce) has a field called `dsnpURI` which contains a uri pointing to a [batch file](/Batches/Overview). 
+1. Query the batch files for the data for a particular DSNP Id to retrieve information about the respective graph. For more on how batch file storage and how to query the batch file see - [batches overview](/Batches/Overview)
+1. Order the retrieved data based on the following  
+    1. Block Number Ascending
+    1. Transaction Index Ascending
+    1. Log Index Ascending
+    1. Batch File Row Index Ascending
```
Comment (amparks100) [inline pages/Graph/Overview.md:74]: Should all these bullet numbers be different too?

**Review by amparks100 — COMMENTED** [review]
This really helps me understand Graphs better! 😃

File: pages/Graph/Overview.md [hunk 4]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+There is no concept of "friendship" within the DSNP social network graph. Friendship requires a mutual acknowledgement between 2 different DSNP Identities. 
+Friendship can be thought as "mutual following" - where 2 DSNP identities are following each other. 
+
+#### Follow
+
+A "follow" is the act of publicly following a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in adding this DSNP Id to a user's social graph.
+
+#### Unfollow
+
+An "unfollow" is the act of publicly unfollowing a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in the removal of this DSNP Id from a user's social graph
```
Comment (enddynayn) [inline pages/Graph/Overview.md:44]: Missing period at the end.

File: pages/Graph/Overview.md [hunk 5]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+All graph change events are signed and added to a [batch file](/Batches/Overview).
+Once that batch file is complete, it is [announced](/Messages/Announce) on the blockchain per the DSNP specification. 
+Each announce event specifies the type of ([dsnpType](/Messages/Announce)) messages that are in the batch file. 
+
+## Graph Retrieval, Ordering & Reading
+To have an accurate representation of the social graph for a given DSNP Identity - it is necessary to retrieve the entire graph from the chain.
+
+To retrieve the graph - do the following:
+1. Retrieve all the log events[dsnpType](/Messages/Types)`GraphChange` from the chain
+1. Retrieve the batch file from each log event. Each log event of type [GraphChange](/Messages/Announce) has a field called `dsnpURI` which contains a uri pointing to a [batch file](/Batches/Overview).
```
Comment (enddynayn) [inline pages/Graph/Overview.md:68]: uri -> URI

File: pages/Graph/Overview.md [hunk 6]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+#### Reading the graph
+Once the retrieval of the entire graph is complete and data is ordered, read the ordered data to see the most recent state of the graph between two respective DSNP Ids (i.e the last graph change event shows Bob has unfollowed Charlie)
+
+If the entire graph has been retrieved and stored, it is possible to then retrieve new events from the point where the last graph ended. This can be done by retrieving `GraphChange` log events from the chain starting with a specific block number. 
+For example: The graph for a given user has been retrieved and stored up until block number 18. A filter can be added to the logs being retrieved from the chain to only retrieve log events starting from block 19.
+
+## Replay Attacks
+
+[Replay attacks](https://en.wikipedia.org/wiki/Replay_attack) are prevented by ensuring that each graph change event is identifiably unique. 
+To allow for uniqueness, each graph change event has a nonce. This nonce field is timestamp represented as microseconds since Unix epoch.
```
Comment (enddynayn) [inline pages/Graph/Overview.md:87]: microseconds? seconds?

File: pages/Graph/Overview.md (hunk 3 above)
Comment (aneyzberg) [reply pages/Graph/Overview.md:74]: markdown will generate consecutive numbers if you follow this format.

File: pages/Graph/Overview.md (hunk 1 above)
Comment (aneyzberg) [reply pages/Graph/Overview.md:24]: great question. Markdown will generate consecutive numbers if you use this format.

**Review by amparks100 — APPROVED** [review]

File: pages/Graph/Overview.md [hunk 7]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+1. Order the retrieved data based on the following  
+    1. Block Number Ascending
+    1. Transaction Index Ascending
+    1. Log Index Ascending
+    1. Batch File Row Index Ascending
+                                                
+For more on ordering see [Message Ordering Specification](/Messages/Ordering).
+
+#### Reading the graph
+Once the retrieval of the entire graph is complete and data is ordered, read the ordered data to see the most recent state of the graph between two respective DSNP Ids (i.e the last graph change event shows Bob has unfollowed Charlie)
```
Comment (acruikshank) [inline pages/Graph/Overview.md:79]: I would title this something like "Creating the Graph" and word this something like:
"Each graph change event represents a state transition for the graph. The state of the graph at any time is given by taking the state of the graph at a previous time and applying all graph change events not previously applied in the order specified above." This describes building the graph from scratch, catching up from a snapshot, and streaming new events from the chain equally well.

File: pages/Graph/Overview.md [hunk 8]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+
+If the entire graph has been retrieved and stored, it is possible to then retrieve new events from the point where the last graph ended. This can be done by retrieving `GraphChange` log events from the chain starting with a specific block number.
```
Comment (acruikshank) [inline pages/Graph/Overview.md:81]: I would drop "from the chain" from this, because it's actually more complicated than that.

File: pages/Graph/Overview.md [hunk 9]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+If the entire graph has been retrieved and stored, it is possible to then retrieve new events from the point where the last graph ended. This can be done by retrieving `GraphChange` log events from the chain starting with a specific block number. 
+For example: The graph for a given user has been retrieved and stored up until block number 18. A filter can be added to the logs being retrieved from the chain to only retrieve log events starting from block 19.
+
+## Replay Attacks
+
+[Replay attacks](https://en.wikipedia.org/wiki/Replay_attack) are prevented by ensuring that each graph change event is identifiably unique. 
+To allow for uniqueness, each graph change event has a nonce. This nonce field is timestamp represented as microseconds since Unix epoch.
+This timestamp will allow a user to detect whether they have seen an event before and can therefore ignore duplicate events thus thwarting the threat of replay attacks.
+
+Without a nonce the following replay attack could happen:
```
Comment (acruikshank) [inline pages/Graph/Overview.md:90]: This is slightly inaccurate. Replay attacks are avoided by strict insistence that clients ignore any event with the same hash (or signature) no matter what the order. The timestamp (probably no need to refer to it as a nonce) allows the user to legitimately redo actions. In the follow, unfollow, follow case without the timestamp, a client can't distinguish between Bob sending the last follow and Alice replaying it, so they must reject it. With the timestamp, Bob's second follow will have a different hash (and different signature).

**Review by acruikshank — APPROVED** [review]
This is good. I have some comments that mostly fall under the "what is the spec trying to achieve category", although I would like to see the timestamp/replay attack issue addressed.

File: pages/Graph/Overview.md (hunk 9 above)
Comment (aneyzberg) [reply pages/Graph/Overview.md:90]: oh that is a good point.

File: pages/Graph/Overview.md (hunk 8 above)
Comment (aneyzberg) [reply pages/Graph/Overview.md:81]: this is true. I was oversimplifying here - but that makes this innacurrate. Good point.

File: .spelling [hunk 10]
```diff
@@ -35,8 +35,10 @@ dequeued
 Docz
 DSNPArchives
 DSNPBatch
+dsnp
```
Comment (wilwade) [inline .spelling:38]:
```suggestion
DSNP
```

File: .spelling [hunk 11]
```diff
@@ -35,8 +35,10 @@ dequeued
...
 dsnpData
 dsnpHash
+dsnpId
```
Comment (wilwade) [inline .spelling:41]: This should all be `dsnpUserId`
```suggestion
```

File: pages/Graph/Overview.md [hunk 12]
```diff
@@ -6,8 +6,12 @@ menu: Graph
 
 # Graph
 
-This specification is intended to describe the social network graph and how we represent it in the protocol.
-This section should also cover how our graph representation could be applied to other networks, i.e. the friendship model vs follower model.
+This specification describes the social network graph and how it is represented in the protocol. 
+
+In this context a [social graph](https://en.wikipedia.org/wiki/Social_graph) means a graph that represents social relations between entities.
+
+The DSNP graph represents users as [DSNP Ids](/Identity/Overview). Each instance of a social network graph is made up of a DSNP Id following other DSNP Ids.
```
Comment (wilwade) [inline pages/Graph/Overview.md:13]: This should be everywhere `DSNP Id` -> `DSNP User Id`
```suggestion
The DSNP graph represents users as [DSNP User Ids](/Identity/Overview). Each instance of a social network graph is made up of a DSNP User Id following other DSNP User Ids.
```

File: pages/Graph/Overview.md (hunk 6 above)
Comment (wilwade) [reply pages/Graph/Overview.md:87]: microseconds is what I remember. Just makes it easier.

File: pages/Graph/Overview.md [hunk 13]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+Without a nonce the following replay attack could happen:
+1. Bob "follows" Charlie and then "unfollows" him.
+ 
+1. Alice takes Bob's follow event and re-announces it as though Bob has decided to follow Charlie again. 
+    - The now reflects a state where Bob is following Charlie - even though this is not the state of the graph that Bob intended.
+
+There is no way to prevent this situation from happening unless each graph change event is identifiably unique. 
+
+## Other Attack Vectors
 TBD
```
Comment (wilwade) [inline pages/Graph/Overview.md:99]:
```suggestion
```

File: pages/Messages/Overview.md [hunk 14]
```diff
@@ -90,6 +90,7 @@ A public follow/unfollow.
 | fromId | DSNP User Id | bytes8 | YES
 | changeType | follow/unfollow| number/enum | YES
 | objectId | Id of followee | bytes8 | YES
+| none | timestamp | number (microseconds since Unix epoch) | YES
```
Comment (wilwade) [inline pages/Messages/Overview.md:93]:
```suggestion
| none | microseconds since Unix epoch | number | no
```

**Review by wilwade — CHANGES_REQUESTED** [review]
Just a few small things as most of the other stuff is covered by others.

File: pages/Graph/Overview.md (hunk 12 above)
Comment (rosalinekarr) [inline pages/Graph/Overview.md:13]:
```suggestion
The DSNP graph represents users as [DSNP Ids](/Identity/Overview).
Each instance of a social network graph is made up of a DSNP Id following other DSNP Ids.
```

File: pages/Graph/Overview.md [hunk 15]
```diff
@@ -6,8 +6,12 @@ menu: Graph
...
 # Graph
 
-This specification is intended to describe the social network graph and how we represent it in the protocol.
-This section should also cover how our graph representation could be applied to other networks, i.e. the friendship model vs follower model.
+This specification describes the social network graph and how it is represented in the protocol. 
+
+In this context a [social graph](https://en.wikipedia.org/wiki/Social_graph) means a graph that represents social relations between entities.
+
+The DSNP graph represents users as [DSNP Ids](/Identity/Overview). Each instance of a social network graph is made up of a DSNP Id following other DSNP Ids.
+When the term "user" is referenced below we are referring to a DSNP Id.
```
Comment (rosalinekarr) [inline pages/Graph/Overview.md:14]: (Not blocking) I think we've been trying to avoid the word "we" in the spec. Maybe we could just say something like, `The term "user" below refers to a DSNP User Id.`

File: pages/Graph/Overview.md [hunk 16]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
 ## Assumptions
+- All graph messages will be of type `GraphChange`.
+- All `GraphChange` messages will be added to batch files for an announcement on block chain
 
-TBD
+##Terminology
+
+#### Friendship
+
+There is no concept of "friendship" within the DSNP social network graph. Friendship requires a mutual acknowledgement between 2 different DSNP Identities.
```
Comment (rosalinekarr) [inline pages/Graph/Overview.md:35]:
```suggestion
There is no concept of "friendship" within the DSNP social network graph.
Friendship requires a mutual acknowledgement between 2 different DSNP Identities. 
```

File: pages/Graph/Overview.md [hunk 17]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+##Terminology
+
+#### Friendship
+
+There is no concept of "friendship" within the DSNP social network graph. Friendship requires a mutual acknowledgement between 2 different DSNP Identities. 
+Friendship can be thought as "mutual following" - where 2 DSNP identities are following each other. 
+
+#### Follow
+
+A "follow" is the act of publicly following a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in adding this DSNP Id to a user's social graph.
```
Comment (rosalinekarr) [inline pages/Graph/Overview.md:40]:
```suggestion
A "follow" is the act of publicly following a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in adding this DSNP User Id to a user's social graph.
```

File: pages/Graph/Overview.md [hunk 18]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+
+#### Unfollow
+
+An "unfollow" is the act of publicly unfollowing a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in the removal of this DSNP Id from a user's social graph.
```
Comment (rosalinekarr) [inline pages/Graph/Overview.md:44]:
```suggestion
An "unfollow" is the act of publicly unfollowing a user (referenced as a (DSNP Identity)[/Identity/Overview]) which results in the removal of this DSNP User Id from a user's social graph.
```

File: pages/Graph/Overview.md [hunk 19]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+
+### Graph
 
-## Details
+The collection of DSNP identities that a given user is following via the DSNP protocol. 
 
+#### Public
+Publicly following or unfollowing a DSNP Id implies that anyone looking for these follow or unfollow events on the blockchain will
+be able to determine who the follower(DSNP Id doing the following) and followee(DSNP Id being followed) are for a given event.
```
Comment (rosalinekarr) [inline pages/Graph/Overview.md:52]:
```suggestion
Publicly following or unfollowing a DSNP Id implies that anyone looking for these follow or unfollow events on the blockchain will be able to determine who the follower(DSNP Id doing the following) and followee(DSNP Id being followed) are for a given event.
```

File: pages/Graph/Overview.md [hunk 20]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+
+## Graph Storage
+All graph change events are signed and added to a [batch file](/Batches/Overview).
+Once that batch file is complete, it is [announced](/Messages/Announce) on the blockchain per the DSNP specification. 
+Each announce event specifies the type of ([dsnpType](/Messages/Announce)) messages that are in the batch file. 
+
+## Graph Retrieval, Ordering & Reading
+To have an accurate representation of the social graph for a given DSNP Identity - it is necessary to retrieve the entire graph from the chain.
+
+To retrieve the graph - do the following:
```
Comment (rosalinekarr) [inline pages/Graph/Overview.md:66]:
```suggestion
To retrieve the graph, do the following:
```

File: pages/Graph/Overview.md (hunk 8 above)
Comment (rosalinekarr) [inline pages/Graph/Overview.md:81]:
```suggestion
If the entire graph has been retrieved and stored, it is possible to then retrieve new events from the point where the last graph ended.
This can be done by retrieving `GraphChange` log events from the chain starting with a specific block number. 
```

File: pages/Graph/Overview.md [hunk 21]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+    1. Log Index Ascending
+    1. Batch File Row Index Ascending
+                                                
+For more on ordering see [Message Ordering Specification](/Messages/Ordering).
+
+#### Reading the graph
+Once the retrieval of the entire graph is complete and data is ordered, read the ordered data to see the most recent state of the graph between two respective DSNP Ids (i.e the last graph change event shows Bob has unfollowed Charlie)
+
+If the entire graph has been retrieved and stored, it is possible to then retrieve new events from the point where the last graph ended. This can be done by retrieving `GraphChange` log events from the chain starting with a specific block number. 
+For example: The graph for a given user has been retrieved and stored up until block number 18. A filter can be added to the logs being retrieved from the chain to only retrieve log events starting from block 19.
```
Comment (rosalinekarr) [inline pages/Graph/Overview.md:82]:
```suggestion
For example: The graph for a given user has been retrieved and stored up until block number 18.
A filter can be added to the logs being retrieved from the chain to only retrieve log events starting from block 19.
```

File: pages/Graph/Overview.md [hunk 22]
```diff
@@ -16,13 +20,81 @@ This section should also cover how our graph representation could be applied to
...
+
+## Replay Attacks
+
+[Replay attacks](https://en.wikipedia.org/wiki/Replay_attack) are prevented by ensuring that each graph change event is identifiably unique. 
+To allow for uniqueness, each graph change event has a nonce. This nonce field is timestamp represented as microseconds since Unix epoch.
```
Comment (rosalinekarr) [inline pages/Graph/Overview.md:87]:
```suggestion
To allow for uniqueness, each graph change event has a nonce.
This nonce field is timestamp represented as microseconds since Unix epoch.
```

**Review by rosalinekarr — CHANGES_REQUESTED** [review]

**Comment by aneyzberg** [comment]
@acruikshank Will you take another look when you have a chance. I took another stab based on what we talked about. Not sure the language is quite there. Happy to adjust again but curious to get your thoughts.

File: pages/Graph/Overview.md [hunk 23]
```diff
@@ -16,13 +21,86 @@ This section should also cover how our graph representation could be applied to
...
+
+This can be done by retrieving `GraphChange` log events starting with a specific block number. 
+For example: The graph for a given user has been retrieved and stored up until block number 18. 
+A filter can be added to the logs being retrieved to only retrieve log events starting from block 19.
 
-TBD
+## Replay Attacks
 
-## Details
+[Replay attacks](https://en.wikipedia.org/wiki/Replay_attack) are prevented by ensuring that each graph change event is identifiably unique.
```
Comment (acruikshank) [inline pages/Graph/Overview.md:96]: This is passive. I would rephrase:
Clients must ignore any GraphChange event that comes after another event with the same signature. This avoids [Replay attacks](https://en.wikipedia.org/wiki/Replay_attack).

File: pages/Graph/Overview.md [hunk 24]
```diff
@@ -16,13 +21,86 @@ This section should also cover how our graph representation could be applied to
...
+A filter can be added to the logs being retrieved to only retrieve log events starting from block 19.
 
-TBD
+## Replay Attacks
 
-## Details
+[Replay attacks](https://en.wikipedia.org/wiki/Replay_attack) are prevented by ensuring that each graph change event is identifiably unique. 
+To allow for uniqueness, each graph change event has a timestamp. 
+This timestamp is represented as microseconds since Unix epoch.
+It allows a user to detect whether they have seen an event before and can therefore ignore duplicate events thus thwarting the threat of replay attacks.
```
Comment (acruikshank) [inline pages/Graph/Overview.md:99]: I'd drop this sentence and the 'also' in the sentence after it. It's the different signatures that allow clients to distinguish duplicate events from new ones.

**Review by acruikshank — APPROVED** [review]
A few stylistic suggestions, but looks good.

**Review by rosalinekarr — APPROVED** [review]

File: pages/Graph/Overview.md [hunk 25]
```diff
@@ -16,13 +21,81 @@ This section should also cover how our graph representation could be applied to
...
+
+An "unfollow" is the act of publicly unfollowing a user (referenced as a (DSNP User Identity)[/Identity/Overview]) which results in the removal of this DSNP User Id from a user's social graph.
+
+### Graph
+
+The collection of DSNP User identities that a given user is following via the DSNP protocol. 
+
+#### Public
+Publicly following or unfollowing a DSNP User Id implies that anyone looking for these follow
+or unfollow events on the blockchain will be able to determine who the follower(DSNP User Id
```
Comment (enddynayn) [inline pages/Graph/Overview.md:54]: I think we need a space after follower

File: pages/Graph/Overview.md [hunk 26]
```diff
@@ -16,13 +21,81 @@ This section should also cover how our graph representation could be applied to
...
+Friendship requires a mutual acknowledgement between 2 different DSNP User Identities. 
+Friendship can be thought as "mutual following" - where 2 DSNP User identities are following each other. 
+
+#### Follow
+
+A "follow" is the act of publicly following a user (referenced as a (DSNP User Identity)[/Identity/Overview]) which results in adding this DSNP User Id to a user's social graph.
+
+#### Unfollow
+
+An "unfollow" is the act of publicly unfollowing a user (referenced as a (DSNP User Identity)[/Identity/Overview]) which results in the removal of this DSNP User Id from a user's social graph.
```
Comment (enddynayn) [inline pages/Graph/Overview.md:46]: (DSNP User Identity)[Identity/Registry] I think we want to point here?

File: pages/Graph/Overview.md [hunk 27]
```diff
@@ -16,13 +21,81 @@ This section should also cover how our graph representation could be applied to
...
+
+The collection of DSNP User identities that a given user is following via the DSNP protocol. 
+
+#### Public
+Publicly following or unfollowing a DSNP User Id implies that anyone looking for these follow
+or unfollow events on the blockchain will be able to determine who the follower(DSNP User Id 
+doing the following) and followee(DSNP User Id being followed) are for a given event.
+
+## Graph Change Announcement Format
+Anytime a user "follows" or "unfollows" someone - a graph change event is created.
```
Comment (enddynayn) [inline pages/Graph/Overview.md:58]: what do you mean by event?

File: pages/Graph/Overview.md [hunk 28]
```diff
@@ -16,13 +21,81 @@ This section should also cover how our graph representation could be applied to
...
+A graph change will contain the following information: `fromId`, `changeType`, `objectId`, `nonce`. See [Messages Overview](/Messages/Overview) for more information on these fields.
+
+## Graph Storage
+All graph change events are signed and added to a [batch file](/Batches/Overview).
```
Comment (enddynayn) [inline pages/Graph/Overview.md:62]: event --> message?

Comment (enddynayn) [reply pages/Graph/Overview.md:62]: perhaps mention the graph DSNPType here.

File: pages/Graph/Overview.md (hunk 27 above)
Comment (aneyzberg) [reply pages/Graph/Overview.md:58]: We can say announcement instead. that might be more in line with the Messages Overview.

File: pages/Graph/Overview.md (hunk 28 above)
Comment (aneyzberg) [reply pages/Graph/Overview.md:62]: I think as above maybe announcement is better here? just like above. More in line with the spec? - https://spec.dsnp.org/Messages/Overview But good call out. 
And I link to the overview types just above - so the intent was to not repeat things here.

File: pages/Graph/Overview.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (shannonwells) [reply pages/Graph/Overview.md]: ahhhh, I never knew that's why we were doing that.  what a great idea :)

**Review by enddynayn — APPROVED** [review]
